### PR TITLE
Fix the update of time step for DL optimizer 

### PR DESCRIPTION
**Pull request by lmoneta**
The update of the time step for the optimizer must be each batch update  and not for each epoch. 
This affects the correction for the bias applied to the computed first and second momentum when using ADam. (see https://root.cern.ch/doc/master/Adam_8h_source.html#l00147 ). 

After this correction the obtained convergence results look better and compatible with what obtained when using when using Keras

### CODE REVIEW

File: tmva/tmva/inc/TMVA/DNN/Optimizer.h [hunk 1]
```diff
@@ -72,8 +72,13 @@ class VOptimizer {
    /*! Increments the global step. */
    void IncrementGlobalStep() { this->fGlobalStep++; }
 
+   void ResetGlobalStep() { this->fGlobalStep = 0; }
```
Comment (stwunsch) [inline tmva/tmva/inc/TMVA/DNN/Optimizer.h]: You add the `ResetGlobalStep` function but you never use it, is this ocrrect?

File: tmva/tmva/src/MethodDL.cxx [hunk 2]
```diff
@@ -1432,8 +1432,10 @@ void MethodDL::TrainDeepNet()
       Log() << "   Start epoch iteration ..." << Endl;
       bool debugFirstEpoch = false;
       bool computeLossInTraining = true;  // compute loss in training or at test time
+      size_t nTrainEpochs = 0;
       while (!converged) {
-         optimizer->IncrementGlobalStep();
+         nTrainEpochs++;
+         //optimizer->ResetGlobalStep();
```
Comment (stwunsch) [inline tmva/tmva/src/MethodDL.cxx]: Here it's outcommented.

File: tmva/tmva/src/MethodDL.cxx [hunk 3]
```diff
@@ -1432,8 +1432,10 @@ void MethodDL::TrainDeepNet()
       Log() << "   Start epoch iteration ..." << Endl;
       bool debugFirstEpoch = false;
       bool computeLossInTraining = true;  // compute loss in training or at test time
+      size_t nTrainEpochs = 0;
```
Comment (stwunsch) [inline tmva/tmva/src/MethodDL.cxx]: So the `nTrainEpochs` replaces now the global step for computing the validation, right?

Comment (lmoneta) [reply tmva/tmva/src/MethodDL.cxx]: I added the function to test the resetting of time steps for every epoch, but it did not make sense to do it. 
I agree the function ResetGlobalSteps is not needed and can  be removed. 

Lorenzo

**Pull request merged**